bgpd/bgp_mpath.c

```diff
@@ -180,6 +180,19 @@ int bgp_path_info_nexthop_cmp(struct bgp_path_info *bpi1,
 			}
 		}
 	}
+	/*
+	 * If both nexthops are same then check
+	 * if they belong to same VRF
+	 */
+	if (!compare) {
+		if (bpi1->extra && bpi1->extra->bgp_orig && bpi2->extra
```

Comment:
This check can be dropped under the existing `if (!compare)` above.

Reply:
Yes, that is correct.
But we do want to bring in another change, which would require this condition to be changed to 
`if(!compare && (bpi1->attr->nh_type != NEXTHOP_TYPE_BLACKHOLE))`
So we are currently leaving it outside.

It is also fine, if we want to drop it inside existing if condition and change it later.

Reply:
These changes are removed from this commit. Since it requires topotest changes. It will be committed later.